collect_app/src/main/java/org/odk/collect/android/widgets/BaseStringWidget.java

```diff
@@ -17,14 +17,47 @@
 package org.odk.collect.android.widgets;
 
 import android.content.Context;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.TypedValue;
 import android.widget.EditText;
+import android.widget.TableLayout;
 
 import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.utilities.ViewIds;
 
 public abstract class BaseStringWidget extends QuestionWidget {
     protected EditText answerText;
 
     public BaseStringWidget(Context context, FormEntryPrompt prompt) {
         super(context, prompt);
+        answerText = new EditText(context);
+        answerText.setId(ViewIds.generateViewId());
+        answerText.setTextSize(TypedValue.COMPLEX_UNIT_DIP, getAnswerFontSize());
+
+        // needed to make long read only text scroll
+        answerText.setHorizontallyScrolling(false);
+        answerText.setSingleLine(false);
```

Comment:
`setSingleLine(false)` is the default behavior, right? Might be clearer just to omit that.